src/app/response-handlers.ts

```diff
@@ -154,3 +99,58 @@ const formatXml = (xml) => {
     }
     return formatted;
 };
+
+export function showResults(results, responseContentType) {
+    getJsonViewer().setValue('');
+    getJsonViewer().getSession().insert(0, results);
+    if (responseContentType) {
+        getJsonViewer().getSession().setMode('ace/mode/' + responseContentType);
+    }
+}
+
+export function insertHeadersIntoResponseViewer(headers: Headers) {
+    if (!headers) {
+        return; // Prevents foreach of undefined error
+    }
+
+    // Format headers
+    const headersArr = [];
+    headers.forEach((headerValue, headerKey) => {
+        headersArr.push(headerKey + ': ' + headerValue);
+    });
+
+    getAceEditorFromElId('response-header-viewer').getSession().setValue('');
+    getAceEditorFromElId('response-header-viewer').getSession().insert(0, headersArr.join('\n'));
+}
+
+export function handleHtmlResponse(results) {
+    showResults(results, 'html');
+}
+
+export function handleJsonResponse(results) {
+    results = JSON.stringify(results, null, 4);
+    showResults(results, 'json');
+}
+
+export function handleXmlResponse(results) {
+    results = formatXml(results);
+    showResults(results, 'xml');
+}
+
+export function handleTextResponse(results) {
+    showResults(results, 'plain_text');
+}
+
+export function isImageResponse(contentType: string) {
+    return contentType === 'application/octet-stream' || contentType.substr(0, 6) === 'image/';
+}
+
+export function getContentType(headers: Headers) {
+    const full = headers.get('content-type');
+    const delimiterPos = full.indexOf(';');
+    if (delimiterPos !== -1) {
+        return full.substr(0, delimiterPos);
+    } else {
+        return full;
+    }
+}
```

Comment:
Why is this code changing at all?

Reply:
This is  a false positive, only whitespace changes